src/main/java/org/mule/extension/email/internal/mailbox/imap/IMAPOperations.java

```diff
@@ -233,5 +234,37 @@ public void moveToFolder(@Connection MailboxConnection connection,
     }
   }
 
+  /**
+   * Counts the emails in the {@code mailboxFolder}.
+   * <p>
+   *
+   * @param connection          The corresponding {@link MailboxConnection} instance.
+   * @param mailboxFolder       Mailbox folder where the emails are.
+   */
+  @Summary("Get the total amount of messages in a specified mailbox folder")
+  @DisplayName("Count messages - IMAP")
+  @Throws(EmailMarkingErrorTypeProvider.class)
+  public int countMessagesImap(@Connection MailboxConnection connection,
+                               @Optional(
+                                   defaultValue = INBOX_FOLDER) @OfValues(MailboxFolderValueProvider.class) String mailboxFolder) {
+    try {
+      Folder defaultFolder = connection.getDefaultFolder();
+      IMAPFolder destinationFolder = (IMAPFolder) defaultFolder.getFolder(mailboxFolder);
+      if (!destinationFolder.exists()) {
+        throw new FolderNotFoundException(destinationFolder);
+      }
+
+      destinationFolder.getMessageCount();
+      return destinationFolder.getMessageCount();
```

Comment:
There seems to be an extra redundant call here to `getMessageCount()`.

Reply:
great thanks!